adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -269,7 +288,7 @@ private AuthenticationResult acquireTokenWithCachedItem(final TokenCacheItem cac
             throws AuthenticationException {
         final AuthenticationResult result = acquireTokenWithRefreshToken(cachedItem.getRefreshToken());
         
-        if (result != null) {
+        if (result != null && !result.isExtendedLifeTimeToken()) {
```

Comment:
I don't understand this, if server returns you 500/503/504, you retry and it fails, you won't get a token back from server, then why we're are doing the check here?